overlord/hookstate/ctlcmd/helpers.go

```diff
@@ -96,14 +108,13 @@ func runServiceCommand(context *hookstate.Context, inst *servicestate.Instructio
 		return err
 	}
 
-	ts, err := servicestateControl(st, appInfos, inst)
+	ts, err := servicestateControl(st, appInfos, inst, context)
```

Comment:
probably merits a comment like:

// passing context so we can ignore self-conflicts with the current change